includes/class-sensei-modules.php

```diff
@@ -205,34 +207,57 @@ public function get_lesson_module_if_exists( $post ) {
 		return $lesson_module;
 	}
 
-	private function render_module_select_for_course( $lesson_course, $lesson_module = 0 ) {
-		// Get the available modules for this lesson's course
-		$modules = $this->get_course_modules( $lesson_course );
+	/**
+	 * Renders the lesson module select input.
+	 *
+	 * @param int|null $course_id
+	 * @param int|null $current_module_id
+	 *
+	 * @return string
+	 */
+	private function render_lesson_module_select_for_course( int $course_id = null, int $current_module_id = null ) {
+		// Get the available modules for this lesson's course.
+		$modules = $course_id ? $this->get_course_modules( $course_id ) : [];
+
+		// Build the HTML.
+		$input_name = 'lesson_module';
 
 		$html  = '';
 		$html .= '<input type="hidden" name="' . esc_attr( 'woo_lesson_' . $this->taxonomy . '_nonce' ) . '" id="' . esc_attr( 'woo_lesson_' . $this->taxonomy . '_nonce' ) . '" value="' . esc_attr( wp_create_nonce( plugin_basename( $this->file ) ) ) . '" />';
 
-		// Build the HTML to output
-		if ( is_array( $modules ) && count( $modules ) > 0 ) {
-			$html .= '<select id="lesson-module-options" name="lesson_module" class="widefat" style="width: 100%">' . "\n";
+		if ( $modules ) {
+			$html .= '<select id="lesson-module-options" name="' . esc_attr( $input_name ) . '" class="widefat" style="width: 100%">' . "\n";
 			$html .= '<option value="">' . esc_html__( 'None', 'sensei-lms' ) . '</option>';
 			foreach ( $modules as $module ) {
-				$html .= '<option value="' . esc_attr( absint( $module->term_id ) ) . '"' . selected( $module->term_id, $lesson_module, false ) . '>' . esc_html( $module->name ) . '</option>' . "\n";
+				$html .= '<option value="' . esc_attr( absint( $module->term_id ) ) . '"' . selected( $module->term_id, $current_module_id, false ) . '>' . esc_html( $module->name ) . '</option>' . "\n";
 			}
 			$html .= '</select>' . "\n";
 		} else {
-			$course_url = admin_url( 'post.php?post=' . urlencode( $lesson_course ) . '&action=edit' );
-
-			/*
-			 * translators: The placeholders are as follows:
-			 *
-			 * %1$s - <em>
-			 * %2$s - </em>
-			 * %3$s - Opening <a> tag to link to the Course URL.
-			 * %4$s - </a>
-			 */
-			$html .= '<p>' . wp_kses_post( sprintf( __( 'No modules are available for this lesson yet. %1$sPlease add some to %3$sthe course%4$s.%2$s', 'sensei-lms' ), '<em>', '</em>', '<a href="' . esc_url( $course_url ) . '">', '</a>' ) ) . '</p>';
+			$html .= '<input type="hidden" name="' . esc_attr( $input_name ) . '" value="">';
```

Comment:
For my own clarification, what is this hidden field used for?

Reply:
This is a common strategy to make a form submit an input with no value. We need this to clear the already selected module.

Consider the following scenario:
* I select a course with modules and the module select box is updated.
* I select a module and save the post.
* I select a course with no modules and save the post. <- At this point, we need to have a `lesson_module` input with no value in order to clear the previously selected module.

Another option would be to add a hook that listens on post save and clear the module, but I find this approach simpler and easier to maintain.